source/daplink/interface/target_flash.c

```diff
@@ -253,16 +300,41 @@ static error_t target_flash_erase_chip(void)
 {
     if (g_board_info.target_cfg){
         error_t status = ERROR_SUCCESS;
-        const program_target_t *const flash = g_board_info.target_cfg->flash_algo;
-
-        status = flash_func_start(FLASH_FUNC_ERASE);
-        if (status != ERROR_SUCCESS) {
-            return status;
-        }
-        if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_chip, 0, 0, 0, 0)) {
-            return ERROR_ERASE_ALL;
+        program_target_t * flash, * start_flash_algo = last_flash_algo;
+        uint8_t index = 0;
+        
+        do{ 
+            flash = last_flash_algo;
+            status = flash_func_start(FLASH_FUNC_ERASE);
+            if (status != ERROR_SUCCESS) {
+                return status;
+            }
+            if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_chip, 0, 0, 0, 0)) {
+                return ERROR_ERASE_ALL;
+            }
+            
+            while (!(g_board_info.target_cfg->extra_flash[index].start == 0 && g_board_info.target_cfg->extra_flash[index].end == 0)) {
+                if(g_board_info.target_cfg->extra_flash[index].flash_algo != 0) { //try additional regions if present
+                    //load each flash algo
+                    status = target_flash_set(g_board_info.target_cfg->extra_flash[index].start); 
+                    if (status != ERROR_SUCCESS) {
+                        return status;
+                    }
+                    index++;
+                    break;
+                }
+                index++;
+            }    
+        } while (flash != last_flash_algo); //call erase on the next flash algo
+        
+        //reset flash algo if changed
+        if (flash != start_flash_algo) {
```

Comment:
You can probably leave this out. The current flash algo will be set correctly on the next erase/program.